Document some local semgrep-agent use cases (#62)
returntocorp/semgrep-docs: docs/integrations.md

it's versatile enough to be used locally
to scan a repository with awareness of its git history.

To locally scan what issues your current branch has
that are not found on the `main` branch,
run the following command:

```
docker run -v $(pwd):/src --workdir /src returntocorp/semgrep-agent:v1 python -m semgrep_agent --config p/r2c-ci --baseline-ref main
```

Another use case is when you want to scan only commits
from the past weeks for new issues they introduced.
This can be done by using a git command
that gets the tip of the current branch two weeks earlier:

```
docker run -v $(pwd):/src --workdir /src returntocorp/semgrep-agent:v1 python -m semgrep_agent --config p/r2c-ci --baseline-ref $(git rev-parse '@{2.weeks.ago}')
```

If you want to compare two commits
and find the issues added between them,
you need to checkout the more recent commit of the two
before running Semgrep CI:

```
git checkout $RECENT_SHA
docker run -v $(pwd):/src --workdir /src returntocorp/semgrep-agent:v1 python -m semgrep_agent --config p/r2c-ci --baseline-ref $OLDER_SHA
```

!!! info
    The above commands all require `docker`
    to be installed on your machine.
    They also use Docker volumes

>     to make your working directory accessible to the container.`--config p/r2c-ci` sets the rules to be scanned,

Reviewer: ```suggestion
    to make your working directory accessible to the container.`--config p/r2c-ci` semgrep rule configuration,
```

Semgrep provides integrations with 3rd party services like Slack, Jira, Defect Dojo, and others. To configure these and learn more, visit [Manage > Notifications](https://semgrep.dev/manage/notifications).

# Commit history

While Semgrep CI is designed
for integrating with various CI providers,
it's versatile enough to be used locally
to scan a repository with awareness of its git history.

> To locally scan what issues your current branch has

Reviewer 2: ```suggestion
To locally scan issues in your current branch
```

that are not found on the `main` branch,
run the following command:

```
docker run -v $(pwd):/src --workdir /src returntocorp/semgrep-agent:v1 python -m semgrep_agent --config p/r2c-ci --baseline-ref main
```

Another use case is when you want to scan only commits
from the past weeks for new issues they introduced.
This can be done by using a git command
that gets the tip of the current branch two weeks earlier:

```
docker run -v $(pwd):/src --workdir /src returntocorp/semgrep-agent:v1 python -m semgrep_agent --config p/r2c-ci --baseline-ref $(git rev-parse '@{2.weeks.ago}')
```

> If you want to compare two commits

Reviewer 2: ```suggestion
To compare two commits
```

and find the issues added between them,

> you need to checkout the more recent commit of the two

Reviewer 2: ```suggestion
checkout the more recent commit of the two
```
